packages/amplify-go-function-runtime-provider/src/localinvoke.ts

```diff
@@ -0,0 +1,125 @@
+import path from 'path';
+import fs from 'fs-extra';
+import portfinder from 'portfinder';
+
+import { InvocationRequest, BuildRequest } from 'amplify-function-plugin-interface';
+import { buildResourceInternal, executeCommand } from './runtime';
+import { MAIN_SOURCE, MAX_PORT, BASE_PORT, BIN_LOCAL, MAIN_BINARY, MAIN_BINARY_WIN } from './constants';
+import execa, { ExecaChildProcess } from 'execa';
+
+// Go typing standards dictating JSON properties with PascalCase format
+type LambdaResult = {
+  Response?: string;
+  Error?: string;
+};
+
+const UNKNOWN_ERROR = 'Unknown error occurred during the execution of the Lambda function';
+
+const buildLocalInvoker = async (context: any) => {
+  const localInvokerDir = path.join(__dirname, '..', 'resources', 'localinvoke');
+  const isWindows = /^win/.test(process.platform);
+  const localInvokeExecutableName = isWindows === true ? MAIN_BINARY_WIN : MAIN_BINARY;
+  const localInvokeExecutablePath = path.join(localInvokerDir, localInvokeExecutableName);
+
+  // Check if we need to build it or it already exists
+  if (!fs.existsSync(localInvokeExecutablePath)) {
+    // Build localinvoker
+    context.print.info('Local invoker binary was not found, building it...');
+    executeCommand(['build', MAIN_SOURCE], true, undefined, localInvokerDir);
+  }
+
+  return {
+    executable: localInvokeExecutablePath,
+  };
+};
+
+const startLambda = (request: InvocationRequest, portNumber: number, lambda: { executable: string; cwd: string }) => {
+  const envVars = request.envVars || {};
+
+  envVars['_LAMBDA_SERVER_PORT'] = portNumber.toString();
+
+  const lambdaProcess: ExecaChildProcess = execa.command(lambda.executable, {
+    env: envVars,
+    cwd: lambda.cwd,
+  });
+
+  return lambdaProcess;
+};
+
+const stopLambda = async (lambdaProcess: ExecaChildProcess) => {
+  try {
+    if (lambdaProcess) {
+      lambdaProcess.cancel();
+
+      await lambdaProcess;
+    }
+  } catch (error) {
+    // Intentionally swallowing process killing errors, as kill is propagated as error from execa, but result
+    // of Lambda invocation still be success
+  }
+};
+
+export const localInvoke = async (request: InvocationRequest, context: any) => {
```

Comment:
any reason this and other functions in this file are consts assigned to anonymous functions instead of just functions? If they are only ever used as functions and not as parameters then I think it's more readable to make them normal functions

Reply:
With the introduction of arrow functions the function keyword is like code small IMHO, I prefer arrow functions over traditional ones, since it works in more cases then traditional ones does not, like inside classes and `this` binding..